integration/tests/access/access_api_test.go

```diff
@@ -0,0 +1,295 @@
+// and should not interfere with each other.
+func (s *AccessAPISuite) TestScriptExecutionAndGetAccounts() {
+	// deploy the test contract
+	txResult := s.deployContract()
+	targetHeight := txResult.BlockHeight + 1
+	s.waitUntilIndexed(targetHeight)
+
+	// Run tests against Access 2, which uses local storage
+	s.testGetAccount(s.an2Client)
+	s.testExecuteScriptWithSimpleScript(s.an2Client)
+	s.testExecuteScriptWithSimpleContract(s.an2Client, targetHeight)
+
+	// Run tests against Access 1, which uses the execution node
+	s.testGetAccount(s.an1Client)
+	s.testExecuteScriptWithSimpleScript(s.an1Client)
+	s.testExecuteScriptWithSimpleContract(s.an1Client, targetHeight)
+
+	// this is a specialized test that creates accounts, deposits funds, deploys contracts, etc, and
+	// uses the provided access node to handle the Access API calls. there is an existing test that
+	// covers the default config, so we only need to test with local storage here.
+	mvp.RunMVPTest(s.T(), s.ctx, s.net, s.accessNode2)
+}
+
+func (s *AccessAPISuite) testGetAccount(client *client.Client) {
+	header, err := client.GetLatestBlockHeader(s.ctx, true)
+	s.Require().NoError(err)
+
+	serviceAddress := s.serviceClient.SDKServiceAddress()
+
+	s.Run("get account at latest block", func() {
+		account, err := client.GetAccount(s.ctx, serviceAddress)
+		s.Require().NoError(err)
+		s.Assert().Equal(serviceAddress, account.Address)
+		s.Assert().NotZero(serviceAddress, account.Balance)
+	})
+
+	s.Run("get account block ID", func() {
+		account, err := client.GetAccountAtLatestBlock(s.ctx, serviceAddress)
+		s.Require().NoError(err)
+		s.Assert().Equal(serviceAddress, account.Address)
+		s.Assert().NotZero(serviceAddress, account.Balance)
+	})
+
+	s.Run("get account block height", func() {
+		account, err := client.GetAccountAtBlockHeight(s.ctx, serviceAddress, header.Height)
+		s.Require().NoError(err)
+		s.Assert().Equal(serviceAddress, account.Address)
+		s.Assert().NotZero(serviceAddress, account.Balance)
+	})
+}
+
+func (s *AccessAPISuite) testExecuteScriptWithSimpleScript(client *client.Client) {
+	header, err := client.GetLatestBlockHeader(s.ctx, true)
+	s.Require().NoError(err)
+
+	script := fmt.Sprintf(simpleScript, 42)
+
+	s.Run("execute at latest block", func() {
+		result, err := client.ExecuteScriptAtLatestBlock(s.ctx, []byte(script), nil)
+		s.Require().NoError(err)
+		s.Assert().Equal(cadence.NewInt(42), result)
+	})
+
+	s.Run("execute at block height", func() {
+		result, err := client.ExecuteScriptAtBlockHeight(s.ctx, header.Height, []byte(script), nil)
+		s.Require().NoError(err)
+		s.Assert().Equal(cadence.NewInt(42), result)
+	})
+
+	s.Run("execute at block ID", func() {
+		result, err := client.ExecuteScriptAtBlockID(s.ctx, header.ID, []byte(script), nil)
+		s.Require().NoError(err)
+		s.Assert().Equal(cadence.NewInt(42), result)
+	})
+}
+
+func (s *AccessAPISuite) testExecuteScriptWithSimpleContract(client *client.Client, targetHeight uint64) {
```

Comment:
Could we add a test that increments the counter value in the counter contract and then executes a script that returns that value with a height before the increase and after, and make sure they are correct? This would validate correct indexing and executing. I can make another PR if you think it's out of scope but I actually think it fits nicely with these e2e tests.

Reply:
addressed in [c8da7ec](https://github.com/onflow/flow-go/pull/4791/commits/c8da7ecca3c6db68ec81ef4e4b37a5de39c7294d)